lib/syskit/network_generation/logger.rb

```diff
@@ -82,6 +82,31 @@ def configure
                 end
             end
 
+            def self.default_log_file_name(logger)
+                if logger.properties.file.nil? || logger.properties.file.empty?
+                    return logger.orocos_name.sub(/_[L|l]ogger/, "")
+                end
+
+                logger.properties.file[/.*(?=\.[0-9]+.log)/]
+            end
+
+            def self.compute_next_log_index(previous_file)
+                if previous_file.nil? || previous_file.empty?
+                    return 0
+                end
+
+                next_index = previous_file[/.*(.[0-9]+.log)/, 1][/[0-9]+/].to_i + 1
+
+                next_index
+            end
+
+            def self.setup_default_logger(logger)
+                index = compute_next_log_index(logger.properties.file)
+                file_name = default_log_file_name(logger)
```

Comment:
Just a reminder, you did not change this line as we discussed.

Reply:
On the most cases I'm using the previous file (instead of the logger name), but I need to use something else when there is no previous file

Reply:
> On the most cases I'm using the previous file (instead of the logger name)

I don't see that code ... where is it ?

If there are two codepaths (there are), you should make two methods. Once the component is configured, file is set. I.e. in `#update_properties` file is empty and in `#rotate_log` it is not.

There is actually a corner case that we don't handle right now. If a logger on which we did call rotate_log is stopped and started again, the index will be reset, since we don't update it in the [orocos.rb setup_default_logger](https://github.com/rock-core/tools-orocosrb/blob/master/lib/orocos/process.rb#L258).

Right now, I'm thinking that the best would be to move the all the code responsible to initialize and modify `file` in the default logger case (from LoggerSupport's `#configure`, orocos.rb as well as compute_next_index and friends) into bundles/common_models/orogen/logger.rb. What do you think ?

Reply:
> If there are two codepaths (there are), you should make two methods. Once the component is configured, file is set. I.e. in #update_properties file is empty and in #rotate_log it is not.

In some cases it is nil (test/network_generation/test_logger.rb:179), not sure why

> Right now, I'm thinking that the best would be to move the all the code responsible to initialize and modify file in the default logger case (from LoggerSupport's #configure, orocos.rb as well as compute_next_index and friends)

It seems great

From "orocos.rb" should we move only the `#setup_default_logger`, or should we include `#default_logger_name`, `#default_log_file_name`, etc?

Reply:
done in [ae5cd10](https://github.com/rock-core/tools-syskit/pull/291/commits/ae5cd109faa505cbeabb7c3abce3cda7258b9155)